utils/collectd/patches/931-snmp6-add-ipv6-statistics.patch

```diff
@@ -0,0 +1,255 @@
+From: Nick Hainke <vincent@systemli.org>
+Date: Mon, 7 Dec 2020 19:29:54 +0100
+Subject: [PATCH] snmp6: add ipv6 statistics
+
+ChangeLog: snmp6 plugin: Add plugin for parsing IPv6 statistics
+
+We would like to have pure ipv6 interface statistics. To get them,
+we parse the snmp6 interface.
+
+Signed-off-by: Nick Hainke <vincent@systemli.org>
+---
+ Makefile.am          |   8 +++
+ README               |   4 ++
+ configure.ac         |   2 +
+ src/collectd.conf.in |   6 ++
+ src/snmp6.c          | 135 +++++++++++++++++++++++++++++++++++++++++++
+ src/types.db         |   2 +
+ 6 files changed, 157 insertions(+)
+ create mode 100644 src/snmp6.c
+
+diff --git a/Makefile.am b/Makefile.am
+index 00947da0..7ea431ea 100644
+--- a/Makefile.am
++++ b/Makefile.am
+@@ -1950,6 +1950,14 @@ TESTS += test_plugin_snmp_agent
+ 
+ endif
+ 
++if BUILD_PLUGIN_SNMP6
++pkglib_LTLIBRARIES += snmp6.la
++snmp6_la_SOURCES = src/snmp6.c
++snmp6_la_CFLAGS = $(AM_CFLAGS)
++snmp6_la_LDFLAGS = $(PLUGIN_LDFLAGS)
++snmp6_la_LIBADD = libignorelist.la
++endif # BUILD_PLUGIN_SNMP6
++
+ if BUILD_PLUGIN_STATSD
+ pkglib_LTLIBRARIES += statsd.la
+ statsd_la_SOURCES = src/statsd.c
+diff --git a/README b/README
+index e42e9c24..cc0f1e5d 100644
+--- a/README
++++ b/README
+@@ -419,6 +419,10 @@ Features
+       network devices such as switches, routers, thermometers, rack monitoring
+       servers, etc. See collectd-snmp(5).
+ 
++    - snmp6
++      Read values from SNMP6 (Simple Network Management Protocol). Supports pure
++      IPv6 interface statistics.
++
+     - statsd
+       Acts as a StatsD server, reading values sent over the network from StatsD
+       clients and calculating rates and other aggregates out of these values.
+diff --git a/configure.ac b/configure.ac
+index bcfb8cf5..1d62cc8f 100644
+--- a/configure.ac
++++ b/configure.ac
+@@ -7158,6 +7158,7 @@ AC_PLUGIN([slurm],               [$with_libslurm],            [SLURM jobs and no
+ AC_PLUGIN([smart],               [$plugin_smart],             [SMART statistics])
+ AC_PLUGIN([snmp],                [$with_libnetsnmp],          [SNMP querying plugin])
+ AC_PLUGIN([snmp_agent],          [$with_libnetsnmpagent],     [SNMP agent plugin])
++AC_PLUGIN([snmp6],               [yes],                       [IPv6 Interface traffic statistics via snmp6])
+ AC_PLUGIN([statsd],              [yes],                       [StatsD plugin])
+ AC_PLUGIN([swap],                [$plugin_swap],              [Swap usage statistics])
+ AC_PLUGIN([synproxy],            [$plugin_synproxy],          [Synproxy stats plugin])
+@@ -7531,6 +7532,7 @@ AC_MSG_RESULT([    infiniband  . . . . . $enable_infiniband])
+ AC_MSG_RESULT([    intel_pmu . . . . . . $enable_intel_pmu])
+ AC_MSG_RESULT([    intel_rdt . . . . . . $enable_intel_rdt])
+ AC_MSG_RESULT([    interface . . . . . . $enable_interface])
++AC_MSG_RESULT([    snmp6 . . . . . . . . $enable_snmp6])
```

Comment:
Minor nitpicking, but is this the correct location in the list?

Reply:
nope, sry I will make the ordering correct!
nitpicking is something good, since code quality becomes better. :P :D